### PR TITLE
FOLLOW UP: Preserve column order using ordered dicts with Table(rows=)

### PR DESCRIPTION
**Pull request by pllim**
Follow up of #8587 based on post-merge review from @taldcroft .

Maintainer's note: If this is re-milestoned beyond 3.2, it needs its own change log, but ideally this should go into 3.2.

### CODE REVIEW

**Comment by bsipocz** [comment]
@pllim - Would you mind adding the example as a regression tests from the comment above?

**Comment by pllim** [comment]
Okay, I'll add when I get the chance.

File: astropy/table/tests/test_init_table.py [hunk 1]
```diff
@@ -505,13 +505,16 @@ def test_init_and_ref_from_dict(table_type, copy):
...
-        rows12 = [row1, row2]
-        row3 = dict([('b', 1), ('a', 0)])
-        row4 = dict([('b', 11), ('a', 10)])
-        rows34 = [row3, row4]
-        t1 = table_type(rows=rows12)
-        t2 = table_type(rows=rows34)
-        assert t1.colnames == ['b', 'a']
-        assert t2.colnames == ['a', 'b']
+    row1 = OrderedDict([('b', 1), ('a', 0)])
+    row2 = {'a': 10, 'b': 20}
```
Comment (pllim) [inline astropy/table/tests/test_init_table.py]: Note here that I changed `row2` a little here to adapt from @taldcroft 's example as I don't want to have too much duplicate code that basically tests for the same stuff.

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**